src/main/js/controls/PairedResult/helpers/helpers.js

```diff
@@ -571,8 +571,15 @@ const clickHandler = (graphContext, control, config, canvasSVG) => (
     canvasSVG
         .selectAll(`.${styles.pairedPoint}[aria-describedby="${item.key}"]`)
         .attr("aria-hidden", legendSelected);
-    const boxPath = d3.selectAll(`.${styles.pairedBox}`);
-    showLine(config, boxPath);
+    const pairedBoxGroupClipPath = `url(#${config.clipPathId})`;
+    const pairedBoxGroup = d3.selectAll(`.${styles.pairedBoxGroup}`);
+    pairedBoxGroup.each(function () {
+        const clipPath = d3.select(this).attr("clip-path");
+        if (clipPath === pairedBoxGroupClipPath) {
+            const boxPath = d3.select(this).selectAll(`.${styles.pairedBox}`);
+            showLine(config, boxPath);
+        }
+    });
```

Comment:
Can you add a comment here? Why we are doing this?

Reply:
Added comment [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/a1db3e9718eb0fc6ca8551da9dc335a49a7d63de). 
The reason for this code change is that :
When there are multiple canvases with paired results in each canvas, selecting a paired result legend item in one canvas was affecting paired results in other canvases ( "the line connecting the high and low values were getting hidden/disappeared in other canvases as well") because earlier we were selecting all .carbon-data-pair elements ( including elements from all other canvases ) and passing them to showLine() method.

So now, we are going to select all .carbon-data-pairs-group elements and iterate through each selection. Here, we check the "clip-path" attribute for each selection and if the clip-path value matches the clip-path value of the particular canvas for which a paired result legend item was clicked, then select only those .carbon-data-pair elements and  pass them to showLine() method. 

This ensures that only that particular canvas's paired results line will be hidden when selecting a paired result legend item high/low.